### PR TITLE
RHEL 7 STIG V1R4

### LINKED ISSUE #1891: [DISA] RHEL-07-010180: Maxrepeat should be 2 or 4?
DISA requirement text:
``When passwords are changed the number of repeating consecutive characters must not be more than four characters.rule``

Check text:
``Configure the operating system to require the change of the number of repeating consecutive characters when passwords are changed by setting the "maxrepeat" option. Add the following line to "/etc/security/pwquality.conf conf" (or modify the line to have the required value): maxrepeat = 2``

### PR DESCRIPTION
**Pull request by jamescassell**
#### Description:

- Update STIG reference to V1R4
- Update stig_overlay correspondingly
- add new oval and references for create_module and finit_module

#### Rationale:

- STIG was out of date

- Fixes #1891

### CODE REVIEW

**Comment by jamescassell** [comment]
Not sure why it failed, but I'm also not familiar with building the project...

**Comment by jamescassell** [comment]
Will attempt to build locally this week. Found the issue with the trailing to commit, but that's not the only issue, as indicated by the failing build with that trailing commit reverted.

The instructions say that the overlay should be entirely generated with code. I ran the code as documented, but I did a git add -p to prevent all the references from becoming xxxx. I used the xccdf from 0.1.38.

Should there ever be manual edits to the overlay? Am I doing it wrong?

**Comment by redhatrises** [comment]
@jamescassell what instructions did you follow? The script shouldn't be changing all references to `xxxx`.

Make sure to also bump the version in `rhel7/CMakeLists.txt'

File: rhel7/profiles/stig-rhel7-disa.profile [hunk 1]
```diff
@@ -215,8 +215,6 @@ selections:
     - audit_rules_file_deletion_events_unlinkat
     - rsyslog_remote_loghost
     - rsyslog_nolisten
-    - install_mcafee_antivirus
```
Comment (redhatrises) [inline rhel7/profiles/stig-rhel7-disa.profile]: @jamescassell why was this removed?

File: rhel7/profiles/stig-rhel7-disa.profile [hunk 2]
```diff
@@ -215,8 +215,6 @@ selections:
...
-    - mcafee_antivirus_definitions_updated
```
Comment (redhatrises) [inline rhel7/profiles/stig-rhel7-disa.profile]: @jamescassell why was this removed?

Comment (jamescassell) [reply rhel7/profiles/stig-rhel7-disa.profile]: The STIG no longer calls out mcafee in particular, but also allows e.g., clamav.

Comment (redhatrises) [reply rhel7/profiles/stig-rhel7-disa.profile]: @jamescassell please undo these changes as the STIG further more does call out McAfee.

**Comment by redhatrises** [comment]
@jamescassell does this fully bring in line with V1R4? If so, the RHEL7 description needs to be updated as well.

**Comment by jamescassell** [comment]
@redhatrises I was following the instructions for updating overlay at: https://github.com/OpenSCAP/scap-security-guide/blob/master/docs/manual/developer_guide.adoc

I didn't check each changed item, and I'm not familiar yet with the ins-and-outs of this project.  I attempted to change obvious items.

Once I get a successful build and everything looks good, I'll rebase this PR.  Is it preferred to open a new PR, or just force-push to the same one?

File: rhel7/profiles/stig-rhel7-disa.profile (hunk 2 above)
Comment (jamescassell) [reply rhel7/profiles/stig-rhel7-disa.profile]: How would we update it to allow both mcafee and clamav?

**Comment by redhatrises** [comment]
> Once I get a successful build and everything looks good, I'll rebase this PR. Is it preferred to open a new PR, or just force-push to the same one?

You can do either. We typically rebase.

**Comment by redhatrises** [comment]
@jamescassell please break out the `finit` and `create` additions. They are separate rules/OVAL.

**Comment by jamescassell** [comment]
@redhatrises do you mean into 2separate commits? It's already separate files.

**Comment by jamescassell** [comment]
Or separate PR?

**Comment by shawndwells** [comment]
> On Apr 8, 2018, at 4:23 PM, redhatrises <notifications@github.com> wrote:
> 
> @redhatrises commented on this pull request.
> 
> In rhel7/profiles/stig-rhel7-disa.profile:
> 
> > @@ -215,8 +215,6 @@ selections:
>      - audit_rules_file_deletion_events_unlinkat
>      - rsyslog_remote_loghost
>      - rsyslog_nolisten
> -    - install_mcafee_antivirus
> -    - mcafee_antivirus_definitions_updated
> Technically use of clamav violates the STIG itself as well as DoD CIO policy which mandated the use of McAfee. Who is the vendor supporting clamav? Is clamav commons criteria certified? Where is the clamav STIG?
> 

Iirc DISA says ClamAV is ok. We can check their content to make sure. 

Since this rule is only enabled in the DISA profile might as well align it closely as possible.

**Comment by redhatrises** [comment]
@jamescassell Need to write the rule or XCCDF for RHEL-07-030821 and RHEL-07-030819.

**Comment by jamescassell** [comment]
@redhatrises no matter what I tried, the create-stig-overlay.py script always replaced all `ruleid=""` references with `ruleid="XXXX"`

**Comment by matejak** [comment]
@jamescassell From the technical point of view, you add, revert, and re-add content in multiple commits. Could you please perform commit squashing (e.g. by running `git rebase -i master`), which will make the commit history much cleaner?

**Comment by jamescassell** [comment]
@matejak I'm working on that now.

**Comment by shawndwells** [comment]
@jamescassell fwiw, I have a *ton* of challenges with organizing commits too. @mpreisler and @redhatrises make fun of me for it =P 

Fortunately you used a local development branch which makes squashing commits easier.

Currently this pull request has 15 commits on it. To squash your 15 commits into one, do the following on your local development branch:

``$ git rebase -i HEAD~15``

After you squash locally you'll need to force push them to GitHub (since you're overwriting the ones attached to this PR):
``$ git push origin stig-v1r4 --force``

**Comment by jamescassell** [comment]
closing in favor of #2728, a rebased version of this one.
